src/ensembl/src/services/storage-service.ts

```diff
@@ -0,0 +1,84 @@
+import windowService, {
+  WindowServiceInterface
+} from 'src/services/window-service';
+import merge from 'lodash/merge';
+
+export enum StorageType {
+  LOCAL_STORAGE = 'localstorage',
+  SESSION_STORAGE = 'sessionstorage'
+}
+
+type PrimitiveValue = string | number | boolean | null | undefined;
+type ArrayValue = PrimitiveValue[] | ObjectValue[];
+type ObjectValue = {
+  [key: string]: PrimitiveValue | ArrayValue | ObjectValue;
```

Comment:
Why is `ObjectValue` used here?

Reply:
Because the value of an object can be an object 😄 

```
{
  foo: 'bar',
  foos: ['foo', 'bar'],
  foobj: { bar: 'quoox' }
}
```

Reply:
Ah makes sense